### PR TITLE
POR-1707 update revision when build fails

### PR DESCRIPTION
**Pull request by ianedwards**
## What does this PR do?

- Call CCP from CLI if build fails to update the status of the revision

### CODE REVIEW

File: api/server/handlers/porter_app/update_app_revision_status.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		return
+	}
+
+	updateStatusReq := connect.NewRequest(&porterv1.UpdateRevisionStatusRequest{
+		ProjectId:      int64(project.ID),
+		AppRevisionId:  appRevisionID.String(),
+		RevisionStatus: statusProto,
+	})
+
+	_, err = c.Config().ClusterControlPlaneClient.UpdateRevisionStatus(ctx, updateStatusReq)
```
Comment (ferozemohideen) [inline api/server/handlers/porter_app/update_app_revision_status.go:97]: nit: might want to just return the object from this update even though it isn't used

**Review by ferozemohideen — APPROVED** [review]

File: api/server/handlers/porter_app/update_app_revision_status.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	// Status is the new status to set for the app revision
+	Status models.AppRevisionStatus `json:"status"`
+	// AppRevisionID is the ID of the app revision to update
+	AppRevisionID string `json:"app_revision_id"`
+}
+
+// UpdateAppRevisionStatusResponse is the response object for the /apps/{porter_app_name}/revisions/{app_revision_id} endpoint
+type UpdateAppRevisionStatusResponse struct{}
+
+func (c *UpdateAppRevisionStatusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```
Comment (d-g-town) [inline api/server/handlers/porter_app/update_app_revision_status.go:45]: add comment

File: api/server/handlers/porter_app/update_app_revision_status.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	ctx, span := telemetry.NewSpan(r.Context(), "serve-update-app-revision-status")
+	defer span.End()
+
+	project, _ := ctx.Value(types.ProjectScope).(*models.Project)
```
Comment (d-g-town) [inline api/server/handlers/porter_app/update_app_revision_status.go:49]: telemetry

File: api/server/router/porter_app.go [hunk 4]
```diff
@@ -949,5 +949,34 @@ func getPorterAppRoutes(
 		Router:   r,
 	})
 
+	// POST /api/projects/{project_id}/clusters/{cluster_id}/apps/metrics -> porter_app.NewUpdateAppRevisionStatusHandler
```
Comment (d-g-town) [inline api/server/router/porter_app.go:952]: not `/apps/metrics`

**Pull request merged**